shared/reducers/teams.tsx

```diff
@@ -79,26 +79,20 @@ export default Container.makeReducer<
     oldLoadingInvites.set(loadingKey, isLoading)
     draftState.teamNameToLoadingInvites.set(teamname, oldLoadingInvites)
   },
-  [TeamsGen.setTeamDetails]: (draftState, action) => {
-    const {teamname, teamID} = action.payload
-    const members = Constants.rpcDetailsToMemberInfos(action.payload.members)
-    const details = mapGetEnsureValue(
-      draftState.teamDetails,
+  [TeamsGen.teamLoaded]: (draftState, action) => {
+    const {teamID, details} = action.payload
+    draftState.teamDetails.set(teamID, details)
+  },
+  [TeamsGen.setTeamVersion]: (draftState, action) => {
+    const {teamID, version} = action.payload
+    draftState.teamVersion.set(
       teamID,
-      Constants.makeTeamDetails({id: teamID, teamname})
+      Constants.ratchetTeamVersion(version, draftState.teamVersion.get(teamID))
```

Comment:
Why is this happening in the UI instead of coming directly from the service?

Reply:
IIRC the service doesn't maintain state about what the team version is, they come in gregor notifications from the server.